ADAL/src/request/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -431,84 +439,139 @@ - (void)requestTokenImpl:(ADAuthenticationCallback)completionBlock
         }
     };
 
-    __block BOOL silentRequest = _allowSilent;
-    
     NSString* telemetryRequestId = [_requestParams telemetryRequestId];
     
     // Get the code first:
     [[MSIDTelemetry sharedInstance] startEvent:telemetryRequestId eventName:MSID_TELEMETRY_EVENT_AUTHORIZATION_CODE];
-    [self requestCode:^(NSString * code, ADAuthenticationError *error)
-     {
-         ADTelemetryAPIEvent* event = [[ADTelemetryAPIEvent alloc] initWithName:MSID_TELEMETRY_EVENT_AUTHORIZATION_CODE
-                                                                        context:_requestParams];
-
-         if (error)
-         {
-             if (silentRequest)
-             {
-                 _allowSilent = NO;
-                 [self requestToken:completionBlock];
-                 return;
-             }
-             
-             ADAuthenticationResult* result = (AD_ERROR_UI_USER_CANCEL == error.code) ? [ADAuthenticationResult resultFromCancellation:_requestParams.correlationId]
-             : [ADAuthenticationResult resultFromError:error correlationId:_requestParams.correlationId];
-             
-             [event setAPIStatus:(AD_ERROR_UI_USER_CANCEL == error.code) ? MSID_TELEMETRY_VALUE_CANCELLED:MSID_TELEMETRY_VALUE_FAILED];
-             [[MSIDTelemetry sharedInstance] stopEvent:_requestParams.telemetryRequestId event:event];
-             completionBlock(result);
-         }
-         else
-         {
+    
+    [self requestCode:^(MSIDWebviewResponse *response, ADAuthenticationError *error) {
+        ADTelemetryAPIEvent* event = [[ADTelemetryAPIEvent alloc] initWithName:MSID_TELEMETRY_EVENT_AUTHORIZATION_CODE
+                                                                       context:_requestParams];
+        
+        if (error)
+        {
+            ADAuthenticationResult *result = (AD_ERROR_UI_USER_CANCEL == error.code) ? [ADAuthenticationResult resultFromCancellation:_requestParams.correlationId]
```

Comment:
Shouldn't we automatically fallback to interactive flow if it's a silent request (AD_PROMPT_AUTO scenario)?

Reply:
This is already in interactive flow. I may be misunderstanding your question, so let's sync offline

Reply:
As settled in the offline discussion, this is strictly broker flow, thus we are fine with not having this logic.